xamarin/Xamarin.Forms
# [Enhancement] Add Visual progress behind feature flag

### Description of Change ###
Add a Visual property to VisualElement that allows us to swap out different renderers based on a Visual Type property

https://github.com/xamarin/Xamarin.Forms/issues/4435

This PR also adds a starting set of Material Renderers as the first use case for this behavior
- Button
- Entry
- Frame (CardView)
- ProgressBar (iOS)

### API Changes ###

Added:

```C#
public class VisualElement
{
		public IVisual Visual
		{
			get { return (IVisual)GetValue(VisualProperty); }
			set { SetValue(VisualProperty, value); }
		}
}

```
```C#
public sealed class ExportRendererAttribute : HandlerAttribute
{
		public ExportRendererAttribute(Type handler, Type target) : this(handler, target, null)
		{
		}
}
// Example Usage
[assembly: ExportRenderer(typeof(Xamarin.Forms.Button), typeof(Xamarin.Forms.Platform.iOS.Material.MaterialButtonRenderer), new[] { typeof(VisualRendererMarker.Material) })]
namespace Xamarin.Forms.Platform.iOS.Material
{
	public class MaterialButtonRenderer : ViewRenderer<Button, MButton>

```




### Platforms Affected ### 


- Core/XAML (all platforms)
- iOS
- Android
- UWP

### Before/After Screenshots ### 
There should be no visual changes to standard Default Renderers 
Here is the view when using Material Visuals

![image](https://user-images.githubusercontent.com/5375137/48643300-9b089100-e99c-11e8-8742-5de12dbed301.png)

![image](https://user-images.githubusercontent.com/5375137/48644475-e96b5f00-e99f-11e8-802a-387f5228cc45.png)


### Testing Procedure ###
- Make sure Visuals are hidden behind the experimental flag correctly
- Make sure all current renderer behavior has been maintained (lots of changes to registrar) so that this change isn't breaking

### PR Checklist ###
